pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java

```diff
@@ -118,19 +118,24 @@ public String toExplainString() {
   protected TransferableBlock getNextBlock() {
     try {
       TransferableBlock block = _sourceOperator.nextBlock();
+      boolean isEarlyTerminated;
       if (block.isSuccessfulEndOfStreamBlock()) {
         // Stats need to be populated here because the block is being sent to the mailbox
         // and the receiving opChain will not be able to access the stats from the previous opChain
         TransferableBlock eosBlockWithStats = TransferableBlockUtils.getEndOfStreamTransferableBlock(
             OperatorUtils.getMetadataFromOperatorStats(_opChainStats.getOperatorStatsMap()));
         sendTransferableBlock(eosBlockWithStats);
+        // when sending an EOS block already, early termination flag is ignored even if receiver has requested it.
+        isEarlyTerminated = false;
       } else {
-        sendTransferableBlock(block);
+        isEarlyTerminated = sendTransferableBlock(block);
```

Comment:
Can be simplified (remove `isEarlyTerminated`)
```suggestion
        if (sendTransferableBlock(block)) {
          earlyTerminate();
        }
```